### PR TITLE
Add catboosting classifier

### PR DESCRIPTION
**Pull request by underscrew**
### Status
**READY**

### Todo list
- [x] Documentation
- [x] Tests

### Description of the changes proposed in the pull request
Add [CatBoost Classifier](https://github.com/catboost/catboost)

### CODE REVIEW

**Comment by matheusfacure** [comment]
The way I see it, CatBoost shines with categorical features. Could you plz include those in your unit tests?

**Comment by underscrew** [comment]
@matheusfacure Sure. Will add it this week..
[LightGBM also supports categorical features](https://lightgbm.readthedocs.io/en/latest/Advanced-Topics.html) . Do you think it is needed there too?

**Comment by matheusfacure** [comment]
If by there you mean your other PR. Yes. Unit tests are nice form of documentation. Also, although Cat-boost deals fine with categorical feature, having it might break something else in your model, so its always good to test.

File: src/fklearn/training/classification.py [hunk 1]
```diff
@@ -207,6 +207,123 @@ def p(new_df: pd.DataFrame, apply_shap: bool = False) -> pd.DataFrame:
...
+            shap_output = {"shap_values": shap_values,
+                           "shap_expected_value": np.repeat(shap_expected_value, len(shap_values))}
+
+            col_dict = merge(col_dict, shap_output)
+
+        return new_df.assign(**col_dict)
+
+    p.__doc__ = learner_pred_fn_docstring("CatBoostClassifier", shap=True)
+
+    log = {'CatBoostClassifier': {
```
Comment (hf-lopes) [inline src/fklearn/training/classification.py]: the dict log keys usually have the same name as the docstring `catboost_classification_learner`

File: src/fklearn/training/classification.py [hunk 2]
```diff
@@ -207,6 +207,123 @@ def p(new_df: pd.DataFrame, apply_shap: bool = False) -> pd.DataFrame:
...
+            shap_expected_value = explainer.expected_value
+
+            shap_output = {"shap_values": shap_values,
+                           "shap_expected_value": np.repeat(shap_expected_value, len(shap_values))}
+
+            col_dict = merge(col_dict, shap_output)
+
+        return new_df.assign(**col_dict)
+
+    p.__doc__ = learner_pred_fn_docstring("CatBoostClassifier", shap=True)
```
Comment (hf-lopes) [inline src/fklearn/training/classification.py]: replace with `catboost_classification_learner`

**Comment by underscrew** [comment]
Done?

**Pull request merged**